x/twap/store_test.go

```diff
@@ -475,3 +480,48 @@ func (s *TestSuite) TestGetAllHistoricalPoolIndexedTWAPs() {
 		})
 	}
 }
+
+func (s *TestSuite) TestAccumulatorOverflow() {
+	maxSpotPrice := gammtypes.MaxSpotPrice
+	tests := map[string]struct {
+		// timeDelta is duration in nano seconds.
+		// we use sdk.Dec here because time.Duration would automatically cap to
+		// time.duration.maxDuration without erroring.
+		timeDelta sdk.Dec
+		panics    bool
+	}{
+		"no overflow": {
+			// 2562047h47m16.854775807s in duration, this is over 292 years.
+			timeDelta: sdk.NewDec(2).Power(128),
+			panics:    false,
+		},
+		"overflow": {
+			timeDelta: sdk.NewDec(2).Power(129),
+			panics:    true,
+		},
+	}
+	for name, test := range tests {
+		s.Run(name, func() {
+			s.SetupTest()
+
+			var accumulatorVal sdk.Dec
+
+			fmt.Println(time.Duration(math.Pow(2, 128)))
+			if test.panics {
+				s.Require().Panics(func() {
+					// accumulator value is calculated via spot price * time delta
+					accumulatorVal = maxSpotPrice.Mul(test.timeDelta)
```

Comment:
Lets use the function?

Reply:
ah wait no, this is your point that its the wrong type since we can't represent the max value in the type